### DISCUSSION TITLE
Autogenerated Username Field: Cannot Save Usernames with Capital Letters

### DISCUSSION THREAD

**Discussion by tobiasiv** [payloadcms/payload#7324] [Feature Requests & Ideas]
### Payload Version

3.0.0-beta.67

### Node Version

20.9.0

### Next.js Version

15.0.0-canary.58

### Describe the Bug

When enabling [login by username](https://github.com/payloadcms/payload/pull/7086) , a new autogenerated username field is created. However, this field does not support storing usernames with capital letters. I suggest saving the username in its original case while comparing it in lowercase during login. This way, the username can be case-insensitive while preserving the original format.

### Reproduction Steps

1. Enable login with username in the user collection:
```js
auth: {
  loginWithUsername: {
    allowEmailLogin: true, 
    requireEmail: true, 
  }
}
```

2. Save username with capital letters (e.g. JohnDoe).
3. The username will be stored in the database in lowercase.

**Comment by JarrodMFlesch** [comment]
Hey @tobiasiv — right now, we have no plans to support case-sensitive usernames, but can I ask what your reasoning is for wanting this? We might be able to be convinced that allowing case-sensitivity via a certain opt-in auth property like `caseSensitivity: true`. Would be a simple PR to make if you wanted to. But for now I will convert this to a discussion because this is operating as it should.

Let me know your thoughts!

You could also add another field to the auth collection such as `displayName` that a user could set if you would like something different.

**Comment by tobiasiv** [comment]
Hey @JarrodMFlesch - Thank you for your response. I think case-sensitive usernames would be helpful because if a user saves their username with capital letters, there is currently no way to display the username exactly as the user saved it. This can be important for maintaining user identity and preference.

While adding a displayName field to the auth collection is a potential workaround, it would be simpler and more intuitive for users if their usernames were preserved with the original capitalization. This could improve user experience by allowing them to see their usernames exactly as they entered them.

**Reply by JarrodMFlesch** [reply]
I would just add a displayName field and then use a collectionBeforeChange hook that checks to see if data.username exists and then set `data.displayName = data.displayName || data.username`.

I think it would open a can of worms right now to allow this. As you would be able to create usernames such as `user1` and `User1` which would be less than ideal.
